test/tsk/hashdb/test_binsrch_index.cpp

```diff
@@ -0,0 +1,94 @@
+#include "tsk/base/tsk_os.h"
+#include "tsk/hashdb/tsk_hashdb_i.h"
+#include "catch.hpp"
+
+void test_hdb_binsrch_idx_init_hash_type_info(
+    const TSK_TCHAR *db_name,
+    TSK_HDB_HTYPE_ENUM htype,
+    int expected_return,
+    int expected_hash_len,
+    const TSK_TCHAR *expected_idx_fname,
+    const TSK_TCHAR *expected_idx_idx_fname)
+{
+    TSK_HDB_BINSRCH_INFO hdb_binsrch_info;
+
+    /**
+     * setup values:
+     * normally, this would be done as a part of opening the hashdb, but not 
+     * for this unit test
+     **/
+    hdb_binsrch_info.hash_type = TSK_HDB_HTYPE_INVALID_ID;
+    // this value is read only in hdb_binsrch_idx_init_hash_type_info
+    hdb_binsrch_info.base.db_fname = (TSK_TCHAR *)db_name;
+    int ret_val = hdb_binsrch_idx_init_hash_type_info(&hdb_binsrch_info, htype);
+
+    REQUIRE(ret_val == expected_return);
+
+    if (expected_return == 0)
+    {
+        // if successful call, do comparisons
+        REQUIRE(hdb_binsrch_info.hash_len == expected_hash_len);
```

Comment:
Can the `REQUIURE`s after this one be checked sensibly if this one fails? If so, this should be a `CHECK` instead of a `REQUIRE`, so that it doesn't prevent seeing the results of later tests. (Same goes for the two `REQUIRE`s after this. `REQUIRE` should be used only when execution must not proceed on failure.)